test/main/test.js

```diff
@@ -9,19 +9,53 @@ QUnit.test( "expect query and multiple issue", function( assert ) {
 
 if ( typeof document !== "undefined" ) {
 
-QUnit.module( "fixture" );
+QUnit.module( "fixture", function( hooks ) {
+	var failure = false,
+		values = [ /* initial value (see unshift below), */ "<b>ar</b>", undefined ],
+		originalValue;
+
+	hooks.before( function() {
+		originalValue = QUnit.config.fixture;
+		values.unshift( originalValue );
+	} );
 
-QUnit.test( "setup", function( assert ) {
-	assert.expect( 0 );
-	document.getElementById( "qunit-fixture" ).innerHTML = "foobar";
-} );
+	// Set QUnit.config.fixture for the next test, propagating failures to recover the sequence
+	hooks.afterEach( function( assert ) {
+		if ( failure ) {
+			assert.ok( false, "prior failure" );
+			QUnit.config.fixture = originalValue;
+		} else {
+			QUnit.config.fixture = values.shift();
+		}
+	} );
+
+	QUnit.test( "setup", function( assert ) {
+		document.getElementById( "qunit-fixture" ).innerHTML = "foo";
+		assert.equal( values.length, 3, "proper sequence" );
+		failure = failure || values.length !== 3;
+	} );
 
-QUnit.test( "basics", function( assert ) {
-	assert.equal(
-		document.getElementById( "qunit-fixture" ).innerHTML,
-		"test markup",
-		"automatically reset"
-	);
+	QUnit.test( "automatically reset", function( assert ) {
+		var contents = document.getElementById( "qunit-fixture" ).innerHTML;
+		assert.equal( contents, "test markup" );
+		assert.equal( values.length, 2, "proper sequence" );
+		failure = failure || values.length !== 2 || contents !== "test markup";
```

Comment:
Both of these `"test markup"` strings should be `originalValue` (I _knew_ I was missing something).